src/ephemeris/set_library_permissions.py

```diff
@@ -28,19 +28,21 @@ def set_permissions(gi, library_id, role_ids):
     total = len(datasets)
     est = total*3/60
     # Give User time to abort
-    log.info('\nSuccess! %d datasets found. Processing can take up to %f min', total, est)
-    if input("Do you want to continue? (y/n) ") == "y":
-        with Progress() as progress:
-            task = progress.add_task("[green]Processing datasets...", total=total)
+    log.info('\nSuccess! %d datasets found. Processing can take up to %0.02f min\n', total, est)
+    if auto:
```

Comment:
I think you're missing the `auto` argument in the function signature

Reply:
yes, sorry